manual typos and formatting inconsistencies

Mo-Gul opened the pull request:
So far fixed some typos and removed double/multiple spaces.

Things that still can be done to harmonize/correct stuff in manual:
- [x] UK spelling <--> US spelling, e.g. `behavior` vs. `behaviour` --> I prefer US spelling
- [x] `section~` <--> `Section~` --> most of the `\ref` prefixes use the latter, so I'd harmonize to that
- [x] e.g. `$x$ coordinate` <--> `$x$-coordinate`
- [x] `\emph{Note:\/}` <--> `\emph{Note:}`
- [x] e.g. `Mineral\"olprodukte` --> replace by UTF-8 character --> `Mineralölprodukte`

Are there any opinions/arguments against that proposed changes? If not, I would apply them one by one.

Mo-Gul committed:
added hints which libraries need to be loaded as well to make the example in `pgfmanual-en-tikz-graphs.tex` work closes issue #755)

Mo-Gul committed:
fixed some typos

Mo-Gul committed:
remove double/multiple spaces

ilayn commented on doc/generic/pgf/text-en/pgfmanual-en-tikz-graphs.tex:
```diff
@@ -243,8 +243,9 @@ \subsubsection{Concept: Chain Groups}
 algorithms from the graph drawing libraries to get a better layout. For
-instance, adding |tree layout| to the above code results in the following
-somewhat more pleasing rendering:
+instance, adding |tree layout| to the above code (and
```
didn't you fix this already in #881 ?

Mo-Gul replied:
@ilayn, yes, I did. Have a look at https://github.com/pgf-tikz/pgf/pull/883#discussion_r446149566 for the reason of that.

hmenke commented on doc/generic/pgf/text-en/pgfmanual-en-base-images.tex:
```diff
@@ -66,7 +66,7 @@ \subsection{Declaring an Image}
     Declares an image, but does not paint anything. To draw the image, use
     |\pgfuseimage{|\meta{image name}|}|. The \meta{filename} may not have an
-    extension.  For \textsc{pdf}, the extensions |.pdf|, |.jpg|, and |.png|
+    extension. For \textsc{pdf}, the extensions |.pdf|, |.jpg|, and |.png|
```
```suggestion
    extension.  For \textsc{pdf}, the extensions |.pdf|, |.jpg|, and |.png|
```
This is the correct way to space sentences, see e.g. [Wikipedia](https://en.wikipedia.org/wiki/Sentence_spacing).  Also good text editors like [Emacs](https://www.gnu.org/software/emacs/manual/html_node/emacs/Sentences.html) or [Vim](http://vimdoc.sourceforge.net/htmldoc/options.html#cpo-J) have shortcuts to navigate sentences and therefore need two spaces to distinguish the end of a sentence from the space after “e.g.“ or “i.e.”.

Please revert this and all other similar changes.

Mo-Gul replied:
In the stated Wikipedia article there isn't stated that this is the correct way to space sentences. Rather there seems to be opposing opinions and for the newer age it seems to tend to using on space. But that is not the point here because in LaTeX we can decide what we want to see in the output and has nothing to do with the spaces in the input. (I think we agree on that.)

But of course that **editors** use them is a very relevant point, especially when the main contributor to this project uses one of them ;)

I know that you stated that point in the past already and we agreed that I will not touch any of the "two spaces" you have written (for the above reason). And I am pretty sure that I haven't.

So just for clarity: If you insist to revert that changes I will do it. But I just want to mention *my* point again which is *consistency*. In the given example above there are two more sentences before the one that is diffed and there is only given 1 space after the end-of-sentence-dot.

So could you live with the current pull request or shall I change it as you wish? (Please note that this was a one-time-search for the two spaces and thus will not happen again.)

hmenke replied:
Any news here?

Mo-Gul replied:
I was waiting for your answer which is still lacking. But I'll take it as "you still wish that I revert the end-of-sentence-dots".
[Below](https://github.com/pgf-tikz/pgf/pull/886#issuecomment-650497371) I asked another question which is also not answered yet. Could you please let me know your opinion, too, so I don't do unnecessary/double work. Thank you very much in advance.

Mo-Gul commented:
@hmenke, any opinions on the other points I want to change in the manual as stated in the first comment in this pull request?

Mo-Gul committed:
typo fixes (closes #894)

hmenke commented:
4 should definitely be addressed.
1 if you have time and really care about it.
2,3,5 seem very unnecessary.

Mo-Gul committed:
revert single space after end-of-sentence dot to two spaces

In a previous commit (hopefully) all remaining instances of two consecutive spaces were removed. Henri *insists* that the ones after an end-of-sentence dot should stay, although the rest of the manual reveals another style chosen by the original author(s).
Therefore change 12 instances back to two spaces.

Mo-Gul committed:
fixed typo: typset --> typeset

Mo-Gul committed:
replace some characters with UTF-8 ones

Mo-Gul committed:
\emph{Note:\/} --> \emph{Note:}

Mo-Gul committed:
behaviour --> behavior

Mo-Gul committed:
section~ --> Section~

Mo-Gul committed:
subsection~ --> Section~

Mo-Gul committed:
$x$ coordinate --> $x$-coordinate

Mo-Gul committed:
`i.e. ` --> `i.e.\ ` and `e.g. ` --> `e.g.\ `

Mo-Gul committed:
Merge branch 'master' into Typos-and-Formatting

Mo-Gul commented on doc/generic/pgf/text-en/pgfmanual-en-library-perspective.tex:
```diff
@@ -188,8 +188,9 @@ \subsection{Defining the perspective}
-    To turn of the perspective in $x$-direction, one must set the $x$ component
+    To turn off the perspective in $x$-direction, one must set the $x$ component
     of |p| to \texttt{0} (e.g.\ |p={(0,a,b)}|, where \texttt{a} and \texttt{b}
+
```
```suggestion
```

Mo-Gul commented on doc/generic/pgf/text-en/pgfmanual-en-oo.tex:
```diff
@@ -37,7 +37,8 @@ \subsection{Overview}
 \emph{constructors}, \emph{attributes}, \emph{objects}, \emph{object
-identities}, and (thanks to Saǒ Živanović) \emph{inheritance} and
+identities}, and (thanks to Sašo Živanović) \emph{inheritance} and
+
```
```suggestion
```

Mo-Gul committed:
Removed spurious empty lines

which somehow sneaked in when resolving conflicts with the GitHub tool.

hmenke commented:
Cherry-picked into https://github.com/pgf-tikz/pgf/commit/9ac32f11fd2f8255a8a74b0994cb9aed5c42a3e7
